refresh with latest schema-salad, proposed CWL v1.2.1 changes

mr-c opened the pull request:
Fixes #27 amongst other things

- `Savable` → `Saveable` typo fix changes some class names
- `default` field type is now explicitly including `File` and `Directory` alongside `Any`
- `hints` is now defined as any of the known requirements, or `Any` (should be a nice simplification)
- the refscope for workflow `outputSource` was corrected from 0 to 1

mr-c committed:
refresh with latest schema-salad, proposed CWL v1.2.1 changes

kinow commented on src/main/java/org/w3id/cwl/cwl1_2/InitialWorkDirRequirement.java:
```diff
@@ -19,7 +19,7 @@
 * Auto-generated interface for <I>https://w3id.org/cwl/cwl#InitialWorkDirRequirement</I><BR>This interface is implemented by {@link InitialWorkDirRequirementImpl}<BR> <BLOCKQUOTE>
  Define a list of files and subdirectories that must be staged by the workflow platform prior to executing the command line tool.
- Normally files are staged within the designated output directory. However, when running inside containers, files may be staged at arbitrary locations, see discussion for `Dirent.entryname`. Together with `DockerRequirement.dockerOutputDirectory` this it possible to control the locations of both input and output files when running in containers. </BLOCKQUOTE>
+ Normally files are staged within the designated output directory. However, when running inside containers, files may be staged at arbitrary locations, see discussion for [`Dirent.entryname`](#Dirent). Together with `DockerRequirement.dockerOutputDirectory` it is possible to control the locations of both input and output files when running in containers. </BLOCKQUOTE>
```
Interesting, maybe the markdown links could be rendered to HTML links in the future, at least for Java. Producing something like `<a href="">{@code Dirent.entryName}</a>` (not sure if `{@code }` can go there, but basically a markdown -> javadoc step). Similarly a markdown -> pydoc, and so it goes.

mr-c replied:
Good idea; see https://github.com/common-workflow-language/cwljava/issues/93 for the follow up

kinow reviewed (approved):
Pending CI, but looks good! :+1:

mr-c committed:
add additional examples

from https://github.com/common-workflow-language/cwl-v1.2/tree/e15d7b77ceecc503382561d04eaf1fc140d80f12/tests
